pseudomonas_analysis/2_identify_generic_genes_pathways_pao1.ipynb

```diff
@@ -0,0 +1,1764 @@
+      "text/plain": [
+       "<Figure size 432x432 with 4 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Get generic genes identified by Crow et. al.\n",
+    "GAPE_filename = params['reference_gene_filename']\n",
+    "ref_gene_col = params['reference_gene_name_col']\n",
+    "ref_rank_col = params['reference_rank_col']\n",
+    "\n",
+    "figure_filename = f\"gene_ranking_{col_to_rank_genes}_pao1.svg\"\n",
+    "\n",
+    "corr, shared_ranking = ranking.compare_gene_ranking(\n",
+    "    summary_gene_ranks,\n",
+    "    GAPE_filename,\n",
+    "    ref_gene_col,\n",
+    "    ref_rank_col,\n",
+    "    figure_filename\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get genes that are highly generic in both\n",
+    "generic_both = shared_ranking[(shared_ranking[\"Rank (simulated)\"]>4500) & (shared_ranking[\"prop DEGs\"]>3500)]\n",
+    "generic_both.to_csv(os.path.join(local_dir, \"SOPHIE_GAPE_generic.tsv\"), sep=\"\\t\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get genes that are highly generic by SOPHIE but not by GAPE\n",
+    "generic_SOPHIE_only = shared_ranking[(shared_ranking[\"Rank (simulated)\"]>4500) & (shared_ranking[\"prop DEGs\"]<2000)]\n",
+    "generic_SOPHIE_only.to_csv(os.path.join(local_dir, \"SOPHIE_generic_only.tsv\"), sep=\"\\t\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Takeaway:**\n",
+    "\n",
+    "* X-axis: gene ranking using SOPHIE (trained on Pseudomonas compendium containing ~1K experiments)\n",
+    "* Y-axis: gene ranking using GAPE (curated set of 73 experiments)\n",
+    "\n",
+    "* Compared to the correlation plot comparing SOPHIE trained on the full pseudomonas compendium vs GAPE experiments [here](2_identify_generic_genes_pathways.ipynb), it looks like there is more noise in the above correlation plot. Looking up the experiment ids associated with GAPE [here](https://github.com/DartmouthStantonLab/GAPE/blob/main/Pa_GSEnum_unzip.rds) it looks like GAPE does contain a mix of PAO1, PA14 and clinical strains. So by subsetting out training compendium to only represent PAO1 patterns, which is adding more noise to the bottom left of the plot as expected (genes that GAPE finds generic are not found by SOPHIE due to our limited representation in our training compendium)"
```

Comment:
It may be helpful to plot the line y=x to show which areas have more GAPE genes or more SOPHIE genes. It may not though, I was just confused that it looked to not be symmetrical at first

Reply:
Let me make of note of this and think about it some more. It might make sense to do.